### PR TITLE
Illuminate manager

### PR DESCRIPTION
**Pull request by dlnilsson**
This is a common approach in Laravel echo system, using the manager class with a facade to handle  multiple drivers (clients). 

However we could leverage the [factory classes used in the symfony bundle](https://github.com/php-http/HttplugBundle/tree/master/ClientFactory) to instantiate the clients if we move them to a third package.

### CODE REVIEW

**Comment by Nyholm** [comment]
Thank you for this PR. 
Im +1 for moving factories to a third package (new or existing). Those have nothing todo with Symfony. 

I think the tests are failing because we miss the ´phpunit.xml.dist`. 

Btw, does this allow us to have two Guzzle6 clients with different set of plugins?

**Comment by dlnilsson** [comment]
> Btw, does this allow us to have two Guzzle6 clients with different set of plugins?

Yes, well you could do something like:

``` php
<?php
// place this in a new serviceprovider or extend HttplugServiceprover
$manager->extend('guzzle6-plugins', function () {
        // add custom configuration, plugins etc..
        return new \Http\Adapter\Guzzle6\Client();
});
```

**Comment by welcoMattic** [comment]
Really cool work! Thank you for it. Actually I thought that I had time to work on it, but things are not happen as we will wish 😉

**Comment by sagikazarmark** [comment]
@Nyholm can you manage this? I have exactly 0 experience with Laravel.

**Comment by sagikazarmark** [comment]
BTW thanks for the work on this.

File: README.md [hunk 1]
```diff
@@ -15,9 +15,62 @@ Via Composer
...
+```
 ## Usage
 
+```php
+<?php
+
+use GuzzleHttp\Psr7\Request;
+
+$httplug = app()->make('httplug');
+$request = new Request('GET', 'http://httpbin.org');
```
Comment (Nyholm) [inline README.md:61]: Let's use the configured factory: 

``` php
$factory = app()->make('httplug.message_factory.default');
$request = $factory->createRequest('GET', 'http://httpbin.org');
```

File: README.md [hunk 2]
```diff
@@ -15,9 +15,62 @@ Via Composer
...
+/**
+ * Send request with another driver
+ */
+$response = $httplug->driver('curl')->sendRequest($request);
+
+// with Facade:
+// Default driver
+Httplug::sendRequest($request);
+// another driver
+Httplug::driver('curl')->sendRequest($request)
```
Comment (Nyholm) [inline README.md]: $response = Httplug::driver('curl')->sendRequest($request)

**Comment by Nyholm** [comment]
I do not have a lot of experience with Laravel either. That's why @danijeel reached out and offered to help. 

I've made some comments. Do 👍  or update the readme if they are okey and I'll be happy to merge this PR. 

Thank you!

**Comment by sagikazarmark** [comment]
@GrahamCampbell you might also be interested in this

**Comment by Nyholm** [comment]
Thank you again @danijeel

**Pull request merged**